core/tests/protractor_utils/CollectionEditorPage.js

```diff
@@ -72,21 +72,17 @@ var CollectionEditorPage = function() {
     await waitFor.elementToBeClickable(
       addExplorationButton, 'Unable to find exploration: ' + query);
 
-    var matched = false;
     var dropdownResultElement = element(
       by.cssContainingText('.dropdown-menu', new RegExp(query)));
-    if (await dropdownResultElement.isPresent()) {
-      await action.click('DropDown Result Element', dropdownResultElement);
-      matched = true;
-    }
-    if (!matched) {
-      // Press Tab to fill in the default result should one appear when
-      // none of the answer matches the given query.
-      await action.sendKeys(
-        'Add Exploration Input', addExplorationInput, protractor.Key.TAB);
-      // If query gets zero result, hitting Tab would not enable the
-      // addExplorationButton.
-    }
+    await waitFor.presenceOf(
+      dropdownResultElement, 'Unable to find exploration: ' + query);
+
+    var matchingSearchResult = element(by.cssContainingText(
+      '.uib-typeahead-match', new RegExp(query)));
+    await waitFor.presenceOf(
+      matchingSearchResult, 'Unable to find search result: ' + query);
```

Comment:
We can keep this if you want to be able to control the error message, but it's not needed for the test logic to work since `action.click` includes a waitFor already

Reply:
Oh, I see! Yeah, I think I'm OK with keeping it.